### ISSUE TITLE
Shader reflection is broken for 64-bit integer types

### ISSUE THREAD

**Issue by jeffnn** [microsoft/DirectXShaderCompiler#2919]
"d3dcommon.h" doesn't contain definitions for 64-bit integers in its D3D_SHADER_VARIABLE_TYPE enum.
Consequently, shader reflection does **this** (falling through to the 32-bit type):

  case hlsl::DXIL::ComponentType::I64:
    cbCompSize = 8;
#ifdef DBG
    OutputDebugStringA("DxilContainerReflection.cpp: warning: component of type 'I64' being reflected as if 'I32'\n");
#endif
  **case hlsl::DXIL::ComponentType::I32:**
    componentType = D3D_SVT_INT;
    m_Name = "int";
    break;

  case hlsl::DXIL::ComponentType::U64:
    cbCompSize = 8;
#ifdef DBG
    OutputDebugStringA("DxilContainerReflection.cpp: warning: component of type 'U64' being reflected as if 'U32'\n");
#endif
  case hlsl::DXIL::ComponentType::U32:
    componentType = D3D_SVT_UINT;
    m_Name = "uint";
    break;

**Comment by BitMD** [comment]
fixed here:
https://github.com/microsoft/DirectXShaderCompiler/pull/2916

**Comment by tex3d** [comment]
This was fixed by #2916. You need some special new `D3D_SHADER_VARIABLE_TYPE` values though, until this makes it into a d3dcommon.h that you can include.

```
// Remove this workaround once d3dcommon.h included has these types
#define ADD_16_64_BIT_TYPES
...
#ifdef ADD_16_64_BIT_TYPES
// Disable warning about value not being valid in enum
#pragma warning( disable : 4063 )
#define D3D_SVT_INT16   ((D3D_SHADER_VARIABLE_TYPE)58)
#define D3D_SVT_UINT16  ((D3D_SHADER_VARIABLE_TYPE)59)
#define D3D_SVT_FLOAT16 ((D3D_SHADER_VARIABLE_TYPE)60)
#define D3D_SVT_INT64   ((D3D_SHADER_VARIABLE_TYPE)61)
#define D3D_SVT_UINT64  ((D3D_SHADER_VARIABLE_TYPE)62)
#endif // ADD_16_64_BIT_TYPES
```
